packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -0,0 +1,347 @@
+   * @param value
+   */
+  public addEnvironmentVariable(name: string, value: string) {
+    this.environmentVariablesResolver.addEnvironmentVariable(name, value);
+  }
+
+  /**
+   * Add custom rule to app
+   *
+   * @param source
+   * @param target
+   * @param status
+   * @param condition
+   */
+  public addCustomRule(source: string, target: string, status?: string, condition?: string) {
+    this.customRulesResolver.addCustomRule(source, target, status, condition);
+  }
+
+  /**
+   * Add domain to app
+   *
+   * @param id
+   * @param props
+   */
+  public addDomain(id: string, props: DomainBaseProps): Domain {
+    const domainProps = {
+      app: this,
+      ...props
+    };
+
+    return new Domain(this, id, domainProps);
+  }
+
+  /**
+   * Add branch to app
+   *
+   * @param id
+   * @param props
+   */
+  public addBranch(id: string, props: BranchBaseProps): Branch {
+    const branchProps = {
+      app: this,
+      ...props
+    };
+
+    return new Branch(this, id, branchProps);
+  }
+}
+
+/**
+ * App Interface
+ */
+export interface IApp extends IResource {
+  /**
+   * App ID
+   *
+   * @attribute
+   */
+  readonly appId: string;
+}
+
+/**
+ * App Properties
+ */
+export interface AppProps {
+  /**
+   * Personal Access token for 3rd party source control system for an Amplify App, used to create webhook
+   * and read-only deploy key. Token is not stored.
+   *
+   * @default - No access token
+   */
+  readonly accessToken?: string;
+
+  /**
+   * Credentials for Basic Authorization for an Amplify App.
+   *
+   * @default - No basic auth config
+   */
+  readonly basicAuthConfig?: BasicAuthConfig;
+
+  /**
+   * BuildSpec for an Amplify App.
+   *
+   * @default - No build spec
+   */
+  readonly buildSpec?: BuildSpec;
+
+  /**
+   * Custom rewrite / redirect rules for an Amplify App.
+   *
+   * @default - No custom rules
+   */
+  readonly customRules?: CustomRule[];
+
+  /**
+   * Description for an Amplify App.
+   *
+   * @default - No description
+   */
+  readonly description?: string;
+
+  /**
+   * Environment variables map for an Amplify App.
+   *
+   * @default - No environment variables
+   */
+  readonly environmentVariables?: EnvironmentVariable[];
+
+  /**
+   * IAM Service Role
+   *
+   * @default - No service role
+   */
+  readonly serviceRole?: IRole;
+
+  /**
+   * Name for the Amplify App.
+   */
+  readonly appName: string;
+
+  /**
+   * OAuth token for 3rd party source control system for an Amplify App, used to create webhook and read-only
+   * deploy key. OAuth token is not stored.
+   *
+   * @default - No oauth token
+   */
+  readonly oauthToken?: SecretValue;
+
+  /**
+   * Repository for an Amplify App.
+   *
+   * Required to be in HTTP URL format: https://github.com/awslabs/aws-cdk
+   * If you use a Git URL, it will error.
+   */
+  readonly repository: string;
+
+  /**
+   * Tag for an Amplify App
+   *
+   * @default - No tags applied.
+   */
+  readonly tags?: Tag[];
```

Comment:
You should use `TagManager`

Reply:
Literally no other resource uses `TagManager` for this, nor is there a PR for adding it in this manner to existing resources.